packages/spear-cli/src/utils/dom.ts

```diff
@@ -315,6 +315,62 @@ function insertComponentSlot(
   }
 }
 
+export function embedAssets(state: State, assets: {[key:string]: string}, nodes: Element[], parent?: Element) {
+  const res = parse("") as Element;
+
+  //nodes.forEach((node) => {
+  for (const node of nodes) {
+    const tagName = node.rawTagName;
+    const isTextNode = node.nodeType === 3;
+
+    if (!isTextNode) {
+      switch(tagName) {
+        case "img": {
+          const imgURL = new URL(node.getAttribute("src"), location.href).href;
+          if (assets[imgURL]) {
+            node.setAttribute("src", `data:image/png;base64,${assets[imgURL]}`);
```

Comment:
How about changing it for each image extension or using `image/*`? 🙏

Reply:
That's true.
I introduced `mime-type` package which distinguish the IANA Mime Type.